auth-api/src/auth_api/services/products.py

```diff
@@ -320,8 +337,12 @@ def approve_reject_parent_subscription(parent_product_code: int, is_approved: bo
         if status == ProductSubscriptionStatus.ACTIVE.value:
             return
 
+        is_reapproved = status == ProductSubscriptionStatus.REJECTED.value and is_approved
```

Comment:
I think this should be refactored into a function?

Reply:
Sounds good I will do that